src/EcomDestinationWriter.cs

```diff
@@ -1045,7 +1046,18 @@ public void Write(Dictionary<string, object> row, Mapping mapping, bool discardD
         {
             if ((columnMapping.SourceColumn != null && row.ContainsKey(columnMapping.SourceColumn.Name)) || columnMapping.HasScriptWithValue)
             {
-                if (columnMapping.HasScriptWithValue)
+                if (mapping.DestinationTable.Name.Equals("EcomStockUnit", StringComparison.OrdinalIgnoreCase) && columnMapping.DestinationColumn.Name.Equals("StockUnitStockLocationId", StringComparison.OrdinalIgnoreCase))
+                {
+                    var stockLocationID = GetStockLocationIdByName(row, columnMapping);
+                    dataRow[columnMapping.DestinationColumn.Name] = stockLocationID;
+                    row[columnMapping.SourceColumn.Name] = stockLocationID;
+                }
+
+                if (mappingColumns.Any(obj => obj.DestinationColumn.Name == columnMapping.DestinationColumn.Name && obj.GetId() != columnMapping.GetId()))
```

Comment:
add null checks as columnMapping.DestinationColumn can be null

Reply:
ok